scripts/HO-example.jl

```diff
@@ -0,0 +1,112 @@
+# # Example 5: Harmonic oscillator
+
+# This is an example calculation of the harmonic oscillator with
+# contact interactions that are treated with first-order perturbation
+# theory.
```

Comment:
It is important to state clearly what the purpose of the example code is. Why should someone who browses to this page bother to read through the example? This might just be a brief explanation (about blocks in the matrix, nodal states, transformation to angular momentum basis...?), or a set of bullet-pointed "learning objectives".